pkg/binding/example_implementing_test.go

```diff
@@ -29,7 +29,7 @@ import (
 type ExMessage json.RawMessage
 
 func (m ExMessage) Structured(b binding.StructuredEncoder) error {
-	return b.SetStructuredEvent(format.JSON, m)
+	return b.SetStructuredEvent(format.JSON, &m)
```

Comment:
Change to use *m is not necessary. json.RawMessage is just a type cast for []byte, and in Go  slices are semantically pointers (actually pointer pairs) - passing or assigning a slice does not copy the underlying data. You need to use the copy() or append() built-in functions to copy the data. So passing/assigning ExMessage by value doesn't copy any data.